apps/root/src/pages/position-detail/components/summary-container/components/swaps/components/timeline/index.tsx

```diff
@@ -604,87 +602,89 @@ const buildModifiedRateAndDurationItem = (positionState: DCAPositionModifiedActi
-              pointer
-              extraText={
-                showPrices && (
-                  <DarkTooltip
-                    title={
-                      showCurrentPrice
-                        ? 'Displaying current value. Click to show value on day of withdrawal'
-                        : 'Estimated value on day of withdrawal'
-                    }
+        <ContainerBox flexDirection="column">
+          <Typography
+            variant="bodySmall"
+            fontWeight={500}
+            color={({ palette: { mode } }) => colors[mode].typography.typo3}
+          >
+            <FormattedMessage description="positionWithdrawWithdrawn" defaultMessage="Withdrawn" />
+          </Typography>
+          <ContainerBox alignItems="center" gap={2}>
+            <TokenIcon token={to} size={5} />
+            <ContainerBox flexDirection="column">
+              <ContainerBox gap={1}>
+                <Typography
+                  variant="body"
+                  fontWeight={700}
+                  color={({ palette: { mode } }) => colors[mode].typography.typo2}
+                >
+                  {formatCurrencyAmount(withdrawn, to)}
+                </Typography>
+                {!!toUsd && (
+                  <Tooltip
+                    title={intl.formatMessage(showCurrentPrice ? currentPriceMessage : prevPriceMessage)}
                     arrow
                     placement="top"
-                    onClick={() => setShouldShowCurrentPrice(!showCurrentPrice)}
                   >
-                    <div>(${showCurrentPrice ? currentToUsd.toFixed(2) : toUsd.toFixed(2)} USD)</div>
-                  </DarkTooltip>
-                )
-              }
-            >
-              <Typography variant="body">{formatCurrencyAmount(BigInt(withdrawn), position.to)}</Typography>
-            </CustomChip>
-            {!!yieldAmount && (
-              <>
-                <FormattedMessage description="positionWithdrawn" defaultMessage="+ yield" />
-                <CustomChip
-                  icon={<ComposedTokenIcon isInChip size={4.5} tokenBottom={position.to} />}
-                  pointer
-                  extraText={
-                    showYieldPrices && (
-                      <DarkTooltip
-                        title={
-                          showCurrentYieldPrice
-                            ? 'Displaying current value. Click to show value on day of withdrawal'
-                            : 'Estimated value on day of withdrawal'
-                        }
-                        arrow
-                        placement="top"
-                        onClick={() => setShouldShowCurrentYieldPrice(!showCurrentYieldPrice)}
+                    <Typography
+                      variant="body"
+                      color={({ palette: { mode } }) => colors[mode].typography.typo3}
+                      onClick={() => setShowCurrentPrice(!showCurrentPrice)}
```

Comment:
We should always update state value using an updater function if it's based on prevState
`() => setShowcurrentPrice((prev) => !prev)`

Reply:
Fixed!